doc/specs/stdlib_string_type.md

```diff
@@ -1254,7 +1254,7 @@ The result is a scalar `string_type` value.
 
 ```fortran
 program demo_to_title
-  use stdlib_string_type, only: string_type, to_title
+  use stdlib_string_type
```

Comment:
Was fine to leave the only specifier. Explicit is better than implicit.

Reply:
> ```
> result = slice(str,8,2,stride=-2)
> ```
> 
> Are the characters now taken in reverse reverse?

Behaviour of the function is similar to this fortran loop:

j = 1
do i = `start`, `end`, `stride`

<< insert the character present at index `i` of `input_string` to index `j` of `output_string` >>

j = j + 1
end do


let's say `input_string` = "abcdefgh"
`start` = 8
`end` = 2
`stride` = -2

I am showing few iterations as the function progresses here:
<loop starts>
function is at index = 8
`output_string` = "h"

function is at index = 6
`output_string` = "hf"

function is at index = 4
`output_string` = "hfd"

function is at index = 2
`output_string` = "hfdb"
<loop ends>

so we get a reversed output "hfdb"

Had the stride been positive 2 (`stride` = 2, `start` = 8, `end` = 2): the above mentioned loop would have executed for 0 number of times i.e. function will never reach any index.

Reply:
would require adding `assignment operator(=)` along with `string_type` and `to_title` because the demo code uses this operator (`=`) in line 1261.